### PR TITLE
added test code

### PR DESCRIPTION
**Pull request by kulrk**
Signed-off-by: kulrk <rohan.1.kulkarni@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/dtm/test_single_process_restart.py [hunk 1]
```diff
@@ -171,3 +170,44 @@ def test_write_during_m0d_restart(self):
...
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=log_file_prefix, queue=que)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+        self.log.info("Step 2: Perform Delete Operations :")
+        proc_del_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                              args=(workload_info, que, True, True, False))
```
Comment (priyankaborawake) [inline tests/dtm/test_single_process_restart.py:197]: You can make the Validate option False here.

File: tests/dtm/test_single_process_restart.py [hunk 2]
```diff
@@ -171,3 +170,44 @@ def test_write_during_m0d_restart(self):
...
+        proc_del_op.start()
+        self.log.info("Step 3 : Perform Single m0d Process Restart During Read Operations")
```
Comment (priyankaborawake) [inline tests/dtm/test_single_process_restart.py:199]: Correct the statement here, during delete operations.

**Review by priyankaborawake — CHANGES_REQUESTED** [review]
Requested minor changes

File: tests/dtm/test_single_process_restart.py (hunk 2 above)
Comment (kulrk) [reply tests/dtm/test_single_process_restart.py]: Done

File: tests/dtm/test_single_process_restart.py (hunk 1 above)
Comment (kulrk) [reply tests/dtm/test_single_process_restart.py]: Fixed

**Review by priyankaborawake — APPROVED** [review]

File: tests/dtm/test_single_process_restart.py [hunk 3]
```diff
@@ -171,3 +170,44 @@ def test_write_during_m0d_restart(self):
 
         self.test_completed = True
         self.log.info("ENDED: Verify READ during m0d restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm0
```
Comment (seagate-sarang-sawant) [inline tests/dtm/test_single_process_restart.py:175]: change marker to dtm.

**Review by seagate-sarang-sawant — APPROVED** [review]

File: tests/dtm/test_single_process_restart.py (hunk 3 above)
Comment (kulrk) [reply tests/dtm/test_single_process_restart.py]: changed marker

File: tests/dtm/test_single_process_restart.py [hunk 4]
```diff
@@ -347,3 +347,46 @@ def test_bkt_creation_ios_after_m0d_restart(self):
...
+                                              args=(workload_info, que, True, False, False))
+        proc_del_op.start()
+
+        self.log.info("Step 3: Perform Single m0d Process Restart During Read Operations")
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=POD_NAME_PREFIX,
+                                            container_prefix=MOTR_CONTAINER_PREFIX,
+                                            process=self.m0d_process, check_proc_state=True)
+        assert_utils.assert_true(resp, "Failure in observed during process restart/recovery")
```
Comment (ApurwaMohite) [inline tests/dtm/test_single_process_restart.py:383]: Remove 'in'. I did it for existing tests.

File: tests/dtm/test_single_process_restart.py [hunk 5]
```diff
@@ -347,3 +347,46 @@ def test_bkt_creation_ios_after_m0d_restart(self):
...
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=log_file_prefix, queue=que)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+        self.log.info("Step 2: Perform Delete Operations :")
+        proc_del_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                              args=(workload_info, que, True, False, False))
```
Comment (ApurwaMohite) [inline tests/dtm/test_single_process_restart.py:374]: Are you sending skipcleanup=False?

**Review by ApurwaMohite — APPROVED** [review]

File: tests/dtm/test_single_process_restart.py [hunk 6]
```diff
@@ -347,3 +347,46 @@ def test_bkt_creation_ios_after_m0d_restart(self):
...
+        proc_del_op.start()
+
+        self.log.info("Step 3: Perform Single m0d Process Restart During Read Operations")
```
Comment (sonalk0209) [inline tests/dtm/test_single_process_restart.py:377]: during Read or Delete?

**Review by sonalk0209 — APPROVED** [review]

File: tests/dtm/test_single_process_restart.py (hunk 6 above)
Comment (kulrk) [reply tests/dtm/test_single_process_restart.py]: fixed

File: tests/dtm/test_single_process_restart.py (hunk 4 above)
Comment (kulrk) [reply tests/dtm/test_single_process_restart.py]: fixed

**Pull request merged**